Hi @xiaoxiang781216, Is this related to any recent changes?

```
====================================================================================
Configuration/Tool: pic32mx7mmb/nsh,CONFIG_MIPS32_TOOLCHAIN_PINGUINOL
2026-06-28 16:44:34
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Disabling CONFIG_MIPS32_TOOLCHAIN_GNU_ELF
  Enabling CONFIG_MIPS32_TOOLCHAIN_PINGUINOL
  Building NuttX...
usbmsc_main.c: In function 'msconn_main':
Error: usbmsc_main.c:584:51: error: 'O_WROK' undeclared (first use in this function)
                            luns[num_luns].flags & O_WROK ? false : true);
                                                   ^
usbmsc_main.c:584:51: note: each undeclared identifier is reported only once for each function it appears in
make[2]: *** [/github/workspace/sources/apps/Application.mk:330: usbmsc_main.c.github.workspace.sources.apps.system.usbmsc.o] Error 1
make[2]: Target 'all' not remade because of errors.
make[1]: *** [Makefile:54: /github/workspace/sources/apps/system/usbmsc_all] Error 2
make[1]: Target 'all' not remade because of errors.
make: *** [tools/LibTargets.mk:248: /github/workspace/sources/apps/libapps.a] Error 2
make: Target 'all' not remade because of errors.
/github/workspace/sources/nuttx/tools/testbuild.sh: line 397: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
  [1/1] Normalize pic32mx7mmb/nsh
/github/workspace/sources/nuttx /github/workspace/sources/nuttx
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
/github/workspace/sources/nuttx
/github/workspace/sources/apps /github/workspace/sources/nuttx
HEAD detached at pull/3567/merge
nothing to commit, working tree clean
/github/workspace/sources/nuttx
Build Attempt 1 of 1
====================================================================================
```

--- a/interpreters/berry/.gitattributes
+++ b/interpreters/berry/.gitattributes
@@ -1,0 +1,1 @@
+0001-Fix-Berry-default-port-for-NuttX.patch whitespace=cr-at-eol,-blank-at-eol

--- a/interpreters/berry/0001-Fix-Berry-default-port-for-NuttX.patch
+++ b/interpreters/berry/0001-Fix-Berry-default-port-for-NuttX.patch
@@ -1,27 +1,57 @@
 diff --git a/default/berry.c b/default/berry.c
-index d6ea5d1..bb95fc7 100644
+index 203773f..0df2a99 100644
 --- a/default/berry.c
 +++ b/default/berry.c
-@@ -22 +22,3 @@
+@@ -19,7 +19,9 @@
+ #endif
+ 
+ /* detect operating system name */
 -#if defined(__linux)
 +#if defined(__NuttX__)
 +    #define OS_NAME   "NuttX"
 +#elif defined(__linux)
-@@ -103 +105 @@
+     #define OS_NAME   "Linux"
+ #elif defined(__unix)
+     #define OS_NAME   "Unix"
+@@ -98,7 +100,7 @@
+ struct arg_opts {
+     int idx;
+     const char *pattern;
 -    const char *optarg;
 +    const char *optvalue;
-@@ -148 +150 @@
+     const char *errarg;
+     const char *src;
+     const char *dst;
+@@ -143,10 +145,10 @@ static int arg_getopt(struct arg_opts *opt, int argc, char *argv[])
+             /* the '?' indicates an optional argument after the option */
+             if (++opt->idx < argc && res != NULL
+                 && res[1] == '?' && *argv[opt->idx] != '-') {
 -                opt->optarg = argv[opt->idx++]; /* save the argument */
 +                opt->optvalue = argv[opt->idx++]; /* save the argument */
-@@ -151 +153 @@
+                 return *res;
+             }
 -            opt->optarg = NULL;
 +            opt->optvalue = NULL;
-@@ -272 +274 @@
+             opt->errarg = arg;
+             return res != NULL ? *res : '?';
+         }
+@@ -267,16 +269,16 @@ static int parse_arg(struct arg_opts *opt, int argc, char *argv[])
+         case 's': args |= arg_s; break;
+         case 'm':
+             args |= arg_m;
 -            opt->modulepath = opt->optarg;
 +            opt->modulepath = opt->optvalue;
-@@ -277 +279 @@
+             break;
+         case '?': return args | arg_err;
+         case 'c':
+             args |= arg_c;
 -            opt->src = opt->optarg;
 +            opt->src = opt->optvalue;
-@@ -281 +283 @@
+             break;
+         case 'o':
+             args |= arg_o;
 -            opt->dst = opt->optarg;
 +            opt->dst = opt->optvalue;
+             break;
+         default:
+             break;

